packages/shared_preferences/shared_preferences_linux/test/shared_preferences_linux_test.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'package:file/memory.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+import 'package:shared_preferences_linux/shared_preferences_linux.dart';
+
+MemoryFileSystem fs = MemoryFileSystem.test();
+
+void main() {
+  setUp(() {});
+
+  tearDown(() {});
+
+  Future<String> _getFilePath() async {
+    var directory = await getApplicationSupportDirectory();
+    return path.join(directory.path, 'shared_preferences.json');
+  }
+
+  _writeTestFile(String value) async {
+    await fs.file(await _getFilePath()).writeAsString(value);
+  }
+
+  Future<String> _readTestFile() async {
+    return await fs.file(await _getFilePath()).readAsString();
+  }
+
+  SharedPreferencesLinux _getPreferences() {
+    var prefs = SharedPreferencesLinux();
+    prefs.fs = fs;
+    return prefs;
+  }
+
+  test('getAll', () async {
```

Comment:
These tests are all failing on CI. E.g.:
```
00:12 +0: getAll
00:12 +0 -1: getAll [E]
  FileSystemException: No such file or directory, path = 'home' (OS Error: No such file or directory, errno = 2)
  package:file/src/backends/memory/common.dart 13:5                       checkExists
  package:file/src/backends/memory/memory_file_system.dart 267:9          _MemoryFileSystem.findNode
  package:file/src/backends/memory/memory_file_system_entity.dart 187:23  MemoryFileSystemEntity.internalCreateSync
  package:file/src/backends/memory/memory_file.dart 56:17                 MemoryFile._doCreate
  package:file/src/backends/memory/memory_file.dart 30:14                 MemoryFile._resolvedBackingOrCreate
  package:file/src/backends/memory/memory_file.dart 270:21                MemoryFile.writeAsBytesSync
  package:file/src/backends/memory/memory_file.dart 294:7                 MemoryFile.writeAsStringSync
  package:file/src/backends/memory/memory_file.dart 283:5                 MemoryFile.writeAsString
  shared_preferences_linux_test.dart 23:41                                main._writeTestFile
```

I think you're just missing creating the directory in the memory filesystem before trying to write into it.

(I'm not sure where `'home'` is coming from; is that a stubbed out path_provider?)